packages/catalog-generator/src/test/java/io/kaoto/camelcatalog/generator/CamelCatalogProcessorTest.java

```diff
@@ -262,13 +239,16 @@ void testGetPatternCatalog() throws Exception {
         var aggregateSchema = processorCatalog.withObject("/aggregate").withObject("/propertiesSchema");
         var aggregationStrategy = aggregateSchema.withObject("/properties").withObject("/aggregationStrategy");
         assertEquals("string", aggregationStrategy.get("type").asText());
-        assertEquals("class:org.apache.camel.AggregationStrategy", aggregationStrategy.get("$comment").asText());
 
         var toDSchema = processorCatalog.withObject("/toD").withObject("/propertiesSchema");
-        var uri = toDSchema.withObject("/properties").withObject("/uri");
+        var uri = toDSchema.withArray("/oneOf").get(1).withObject("/properties").withObject("/uri");
         assertEquals("string", uri.get("type").asText());
-        var parameters = toDSchema.withObject("/properties").withObject("/parameters");
-        assertEquals("object", parameters.get("type").asText());
+        assertFalse(toDSchema.withArray("/oneOf").get(1).withObject("/properties").has("/parameters"));
+
+        var toSchema = processorCatalog.withObject("/to").withObject("/propertiesSchema");
+        var toUri = toSchema.withArray("/oneOf").get(1).withObject("/properties").withObject("/uri");
+        assertEquals("string", toUri.get("type").asText());
+        assertFalse(toSchema.withArray("/oneOf").get(1).withObject("/properties").has("/parameters"));
```

Comment:
I think we might need to revert this one, because `to`, `toD` and `poll` needs to be a single schema.

Reply:
Not exactly, let's catch up tomorrow and I'll show you the issue that we're facing with these 3 EIPs